File: useless.erl
-module(useless).
-define(sub(X,Y), X-Y).
-export([add/2, hello/0, greet_and_add_two/1, call_macro/2]).

add(A, B) ->
    A + B +1.

hello() ->
	io:format("Hello, world!~n").

greet_and_add_two(X) ->
    hello(),
    add(X,2).

call_macro(X,Y) ->
	?sub(X,Y).

File: useless.ex
defmodule Useless do
  def add(a, b) do
    a + b
  end

  def hello() do
	  IO.puts("Hello, world!")
	end

  def greet_and_add_two(x) do
    hello()
    add(x,2)
  end

  defmacro sub(x, y) do
    quote do
      unquote(x) - unquote(y)
    end
  end

  def call_macro(x, y) do
    sub(x, y)
  end
end
